src/altinn-app-frontend/src/components/summary/InputSummaryBoilerplate.tsx

```diff
@@ -0,0 +1,21 @@
+import * as React from 'react';
+
+interface BoilerplateProps extends React.PropsWithChildren<any> {
+  formData: any;
+  setDisplayData: (v: any) => void;
+}
+
+export default function InputSummaryBoilerplate({ setDisplayData, formData, children }: BoilerplateProps) {
```

Comment:
This approach works fine, but I wanted to mention custom hooks as an alternative 🙂
(I wrote the hook directly here, so its not tested and may typos or other issues - so if you change to this, be sure to check that it works as expected 😅)

```jsx
const useDisplayData = ({formData}) => {
  const [displayData, setDisplayData] = React.useState('');

  React.useEffect(() => {
    if (formData && typeof formData === 'object') {
      let displayString = '';
      Object.keys(formData).forEach((key, index) => {
        displayString += `${index > 0 ? ' ' : ''}${formData[key]}`;
      });
      setDisplayData(displayString);
    } else {
      setDisplayData(formData);
    }
  }, [formData, setDisplayData]);

  return {displayData}
}
```

And usage would be
```jsx

function GroupInputSummary({ formData, label }: ISingleInputSummary) {
  const classes = useStyles();

  const {displayData} = useDisplayData({formData});

  return (
      <Typography variant='body1'>
        <span className={classes.label}>
          {label} {': '}
        </span>
        <span>{displayData}</span>
      </Typography>
    )
}
```

Custom hooks is a nice way to reuse logic across components that doesnt have rendered output. We avoid the wrapping components and having to pass down extra props or state handlers etc..

Reply:
Nice